### PR TITLE
Add saving

### LINKED ISSUE #79: As a user I can save flashcards to storage file
... so that flashcards don't disappear when I close the program

### PR DESCRIPTION
**Pull request by quinnyyy**
The functionality for saving is here. I haven't updated the tests and javadoc yet. I will do that tomorrow or the next day. But I wanted to get this PR up to give a chance to take a look at code quality.

The overview of this change (so that it is easier to read) is as follows:
* Create a storage class to do saving/deletion. Using 3rd party json library to save.
* Create a Savable interface. Flashcards and groups implement this interface so they can be saved.
* Add another abstract class, ModifyingCommand, that inherits from Command and is extended by Commands that modify something so they need to deal with file I/O after they run
* Refactor AddFlashcardToGroupCommand. I moved the logic from GroupList to AddFlashcardToGroupCommand. This makes it more modular so that we can easily save the group. 

resolves #79 , #90 , #99

### CODE REVIEW

File: src/main/java/seedu/tp/commands/Command.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package seedu.tp.commands;
 
 import seedu.tp.exceptions.HistoryFlashcardException;
+import seedu.tp.storage.Savable;
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/commands/Command.java:4]: Hmmm why do we import `Savable` here?

**Review by Zhilin-Huang — COMMENTED** [review]
Generally looks good to me! But still need to resolve the merge conflicts

**Comment by quinnyyy** [comment]
I made a CommandFeedback class that command.execute() can return which contains the feedback to the user

**Review by Zhilin-Huang — APPROVED** [review]
LGTM! Thanks for putting in so much work

**Comment by quinnyyy** [comment]
> LGTM! Thanks for putting in so much work

no u

**Pull request merged**